main.go

```
@@ -700,56 +977,140 @@ func (p *ProxyService) waitForWake(ctx context.Context, target *TargetState) err
+		}
+	}
+
+	if cached, reason := p.readyCacheStatus(route); !cached {
+		p.logger.Info("Route %s not ready (%s), waiting before forwarding", route.Name, reason)
+		if err := p.waitForReady(ctx, route); err != nil {
+			p.logger.Error("Route %s did not become ready: %v", route.Name, err)
+			return
+		}
+	}
+
+	upstream, err := net.Dial("tcp", route.Destination)
```

golangci-lint (2.12.2)

[error] net.Dial must not be called. use (*net.Dialer).DialContext

(noctx)